pwndbg/commands/ipython_interactive.py

```diff
@@ -0,0 +1,42 @@
+"""
+Command to start an interactive IPython prompt.
+"""
+import sys
+from contextlib import contextmanager
+
+import gdb
+
+import pwndbg.commands
+
+
+@contextmanager
+def switch_to_ipython_env():
+    """We need to change stdout/stderr to the default ones, otherwise we can't use tab or autocomplete"""
+    # Save GDB's stdout and stderr
+    saved_stdout = sys.stdout
+    saved_stderr = sys.stderr
+    # Use Python's default stdout and stderr
+    sys.stdout = sys.__stdout__
+    sys.stderr = sys.__stderr__
+    yield
+    # Restore GDB's stdout and stderr
+    sys.stdout = saved_stdout
+    sys.stderr = saved_stderr
+    # Restore Python's default ps1 and ps2 for GDB's `pi` command
+    sys.ps1 = ">>> "
+    sys.ps2 = "... "
+
+
+@pwndbg.commands.ArgparsedCommand("Start an interactive IPython prompt.")
+def ipi():
+    with switch_to_ipython_env():
+        # Use `gdb.execute` to embed IPython into GDB's variable scope
+        code4ipython = """import IPython
```

Comment:
Do we need to add ipython or jedi into requirements?

Reply:
Hmm, maybe add `ipython` into requirements is enough because `ipython` depends on `jedi`.

But how about we do something similar like how we treat `ipdb`?

For example:

```python
try:
    import IPython
except ImportError:
    print some hint message to user?
```

Which one do you prefer more?

Reply:
Let's make this opt-in: let's inform users to install ipython if they want to use this.